lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -460,6 +504,130 @@ def _check_job_arguments(self, jobs, has_parent_job=False):
             # although most likely jobs aren't operating on the same object
             self._check_ws_objects(source_objects=job.get(_SOURCE_WS_OBJECTS))
 
+    def retry_multiple(self, job_ids, as_admin=False):
+        """
+        #TODO Add new job requirements/cgroups as an optional param
+        :param job_ids: The list of jobs to retry
+        :param as_admin: Run with admin permission
+        :return: The child job ids that have been retried or errors
+        """
+        retried_jobs = []
+        for job_id in job_ids:
+            try:
+                child_job_id = self.retry(job_id, as_admin=as_admin)
+                retried_jobs.append({"job_id": child_job_id})
+            except Exception as e:
+                retried_jobs.append({"job_id": None, "error": f"{e}"})
+        return retried_jobs
+
+    def retry(self, job_id, as_admin=False) -> str:
+        """
+        #TODO Add new job requirements/cgroups as an optional param
+        :param job_id: The main job to retry
+        :param as_admin: Run with admin permission
+        :return: The child job id that has been retried
+        """
+        job = self.sdkmr.get_job_with_permission(
+            job_id, JobPermissions.WRITE, as_admin=as_admin
+        )  # type: Job
+
+        # Cannot retry a retried job, you must retry the parent
+        if job.retry_parent:
+            raise CannotRetryARetryException(
+                f"Cannot retry a retried job attempt, please retry the parent {job.retry_parent}"
+            )
+
+        # Get run job params from db, and inject parent job id, then run it
+        run_job_params = self._get_run_job_params_from_existing_job(
+            job, user_id=self.sdkmr.user_id
+        )
+
+        # TODO verify only parents have a batch_job, maybe rename to batch_job_parent?
+        if run_job_params.get("batch_job"):
+            # Otherwise we have to make this endpoint always return a list of retried_jobs?
+            raise Exception("Use batch retry endpoint for this job. ")
+
+        # Cancel job and it's children
+        self.sdkmr.cancel_job(
+            job_id=job_id,
+            terminated_code=TerminatedCode.terminated_by_user_retry.value,
+            as_admin=as_admin,
+        )
+
+        # Submit job to job scheduler or fail and not count it as a retry attempt
+        run_job_params[_PARENT_RETRY_JOB_ID] = job_id
+        child_job_id = self.run(params=run_job_params, as_admin=as_admin)
+
+        # Save that the job has been retried, and increment the count
+        # TODO Use and create a method in sdkmr?
+        job.modify(inc__retry_count=1, set__retried=True)
+
+        # Should we compare the original and child job to make sure certain fields match,
+        # to make sure the retried job is correctly submitted? Or save that for a unit test?
```

Comment:
you could do something like create a hash from the params and compare between the two jobs.

Reply:
Is that something that should be done here, or in a  test? (Currently doing it in a test)